ledger-tool/src/main.rs

```diff
@@ -3974,25 +3974,13 @@ fn main() {
                     );
                     exit(1);
                 }
-                let ancestor_iterator = AncestorIterator::new(start_root, &blockstore)
-                    .take_while(|&slot| slot >= end_root);
-                let roots_to_fix: Vec<_> = ancestor_iterator
-                    .filter(|slot| !blockstore.is_root(*slot))
-                    .collect();
-                if !roots_to_fix.is_empty() {
-                    eprintln!("{} slots to be rooted", roots_to_fix.len());
-                    for chunk in roots_to_fix.chunks(100) {
-                        eprintln!("{chunk:?}");
```

Comment:
One negative thing of using the existing function is this output that indicates something happened is gone. This log line does get emitted which might be passable:
```
[2023-06-28T20:19:33.963267210Z INFO  solana_metrics::metrics] datapoint: blockstore-scan_and_fix_roots find_missing_roots_us=7338i num_roots_to_fix=1194i fix_roots_us=544i
```

An alternative would be making `scan_and_fix_roots()` return the number of roots it repaired, and then printing that out in ledger-tool with something like:
```rust
eprintln!("Successfully repaired {num_repaired_roots} roots");
```

Reply:
Done